decoder.go

```diff
@@ -561,18 +570,26 @@ func (d *decoder) tryUnmarshalers(key string, value reflect.Value, def string) (
 		// Unmarshal corresponding json.
 		return true, errorWithKey(t.UnmarshalJSON(b), key)
 	case encoding.TextUnmarshaler:
+		if skip(&v, def) {
```

Comment:
because it is a type switch, and we want to return false in case if a type is not one of the unmarshalers